tests/csv.test

```diff
@@ -111,6 +111,25 @@ $  printf 'fields date, description, amount, balance1, balance2\ndate-format %%d
 >=0
 
 
+# 11. More than two postings 
+$  printf 'fields date, description, amount, balance1, balance2, amount3,comment3\ndate-format %%d/%%Y/%%m\ncurrency $\naccount1 assets:myacct\naccount3   expenses:tax\n' >t.$$.csv.rules; printf '10/2009/09,Flubber Co,50,321,123,0.234,VAT\n' | hledger -f csv:- --rules-file t.$$.csv.rules print && rm -rf t.$$.csv.rules
+2009/09/10 Flubber Co
+    assets:myacct               $50 = $321
+    expense:unknown     = $123
+    expenses:tax             $0.234  ; VAT
+
+>=0
+
+# 12. More than two postings and different currencies 
+$  printf 'fields date, description, amount, balance1, balance2, currency3, amount3,comment3\ndate-format %%d/%%Y/%%m\ncurrency $\naccount1 assets:myacct\naccount3   expenses:tax\n' >t.$$.csv.rules; printf '10/2009/09,Flubber Co,50,321,123,£,0.234,VAT\n' | hledger -f csv:- --rules-file t.$$.csv.rules print && rm -rf t.$$.csv.rules
+2009/09/10 Flubber Co
+    assets:myacct               $50 = $321
+    expense:unknown     = $123
+    expenses:tax             £0.234  ; VAT
+
+>=0
+
+
```

Comment:
Looking good!

I wonder how we can write these tests so that the other input files' content is more visible. They are hard to read.

How well does this handle the various validity checks I (hope I) mentioned on the original issue ? Eg requiring the numbers to be consecutive, starting at 1, etc. etc. ?

Reply:
https://github.com/simonmichael/hledger/issues/627#issuecomment-342315800 -> “Some things to check..”
Maybe total validity checking isn’t important, I’m just hoping for good error messages and easy troubleshooting so this doesn’t cause confusion.

Reply:
> I wonder how we can write these tests so that the other input files' content is more visible. They are hard to read.

I've pushed test 16 that does away with the use of printf and allows one to lay out csv file and rules file on several lines with no extra quoting/escaping. What do you think?

> How well does this handle the various validity checks I (hope I) mentioned on the original issue ? Eg requiring the numbers to be consecutive, starting at 1, etc. etc. ?

I think that there is no need for this. Csv parser should not attempt to be holier than thou - if input+rules produces output that is impossible to balance, then it is on the user to fix the rules or fix the data. The parser itself should be as small as possible - any validity checks would just introduce opportunities to make subtle bugs that will prevent csv parsing. Plus validity check would have to always be in line with what Journal parser actually allows (or not), and as Journal evolves I see no easy way to ensure this.

Eventually, validity checks will get out of sync with journal parser and will start to produce false rejects. 

If users want to immediately validate the output of csv parser, they could just pipe it into "hledger -f - print", and we could mention this in the docs and that is it.

Reply:
> I wonder how we can write these tests so that the other input files' content is more visible. They are hard to read.

I've pushed test 16 that does away with the use of printf and allows one to lay out csv file and rules file on several lines with no extra quoting/escaping. What do you think? If you like it, i will update the rest of the tests to match